contracts/src/spaces/entitlements/rule/RuleEntitlementV2.sol

```diff
@@ -38,17 +37,20 @@ contract RuleEntitlementV2 is
   UUPSUpgradeable,
   IRuleEntitlementV2
 {
-  using EnumerableSet for EnumerableSet.Bytes32Set;
+  // keccak256(abi.encode(uint256(keccak256("spaces.entitlements.rule.storage.v2")) - 1)) & ~bytes32(uint256(0xff))
+  bytes32 private constant STORAGE_SLOT =
+    0x858b72ffb9b2fa0fc89266b0dd2710729cbe0194d0dc7ad7f830ebf836219000;
 
   struct Entitlement {
     address grantedBy;
     uint256 grantedTime;
-    RuleData data;
+    bytes data;
   }
 
-  mapping(uint256 => Entitlement) internal entitlementsByRoleId;
-
-  address public SPACE_ADDRESS;
+  struct Layout {
```

Comment:
Why not annotate `Layout` with `@custom:storage-location` and bump the solc version to ^0.8.20 as erc7201 describes?